TidepoolOnboardingKitUI/View Models/OnboardingViewModel.swift

```diff
@@ -7,47 +7,80 @@
 //
 
 import Foundation
+import Combine
 import LoopKit
 import LoopKitUI
 
-class OnboardingViewModel: ObservableObject, OnboardingNotifying, CGMManagerCreateNotifying, CGMManagerOnboardNotifying, PumpManagerCreateNotifying, PumpManagerOnboardNotifying, ServiceCreateNotifying, ServiceOnboardNotifying, CompletionNotifying {
-    weak var onboardingDelegate: OnboardingDelegate?
+class OnboardingViewModel: ObservableObject, CGMManagerCreateNotifying, CGMManagerOnboardNotifying, PumpManagerCreateNotifying, PumpManagerOnboardNotifying, ServiceCreateNotifying, ServiceOnboardNotifying {
     weak var cgmManagerCreateDelegate: CGMManagerCreateDelegate?
     weak var cgmManagerOnboardDelegate: CGMManagerOnboardDelegate?
     weak var pumpManagerCreateDelegate: PumpManagerCreateDelegate?
     weak var pumpManagerOnboardDelegate: PumpManagerOnboardDelegate?
     weak var serviceCreateDelegate: ServiceCreateDelegate?
     weak var serviceOnboardDelegate: ServiceOnboardDelegate?
-    weak var completionDelegate: CompletionDelegate?
 
-    private let cgmManagerProvider: CGMManagerProvider
-    private let pumpManagerProvider: PumpManagerProvider
-    private let serviceProvider: ServiceProvider
+    let onboardingProvider: OnboardingProvider
 
-    @Published var isWelcomeComplete: Bool = false
+    @Published var isWelcomeComplete: Bool
+    @Published var isTherapySettingsComplete: Bool
+    @Published var therapySettings: TherapySettings?
 
-    init(cgmManagerProvider: CGMManagerProvider, pumpManagerProvider: PumpManagerProvider, serviceProvider: ServiceProvider) {
-        self.cgmManagerProvider = cgmManagerProvider
-        self.pumpManagerProvider = pumpManagerProvider
-        self.serviceProvider = serviceProvider
+    private lazy var cancellables = Set<AnyCancellable>()
+
+    init(onboarding: TidepoolOnboardingUI, onboardingProvider: OnboardingProvider) {
+        self.onboardingProvider = onboardingProvider
+
+        self.isWelcomeComplete = onboarding.isWelcomeComplete
+        self.isTherapySettingsComplete = onboarding.isTherapySettingsComplete
+        self.therapySettings = onboarding.therapySettings
+
+        $isWelcomeComplete
+            .dropFirst()
```

Comment:
why `dropFirst()`?  Not a blocker, just a question -- it isn't obvious to me...

Reply:
As soon as you connect the `sink` here it fires off the current state of the local variable here. Since we *just* read that local variable from destination in `sink`, the `dropFirst` prevents that unnecessary write. (And, since the write causes the onboarding state to serialize, it also prevents that unnecessary serialization.)

Reply:
I thought that was the case.  Why not remove the code above that reads it and remove the `dropFirst()` then?  Seems unnecessary to me...

Reply:
Because we need to set the initial state in the view model from the persisted onboarding. This covers the case where the user force-quits the app in the middle of onboarding. The view model is transient (only while the onboarding UI displayed), but the onboarding (an instance of `TidepoolOnboardingUI` here) maintains the persisted state in `UserDefaults` (just like the other managers and services). So, when onboarding is resumed, the instance of `TidepoolOnboardingUI` is rehydrated, the UI launched, and then the view model is instantiated based upon the newly rehydrated instance of `TidepoolOnboardingUI`.

Reply:
Ok.  I understand the desired behavior, but not exactly sure why that is related to my code suggestion.  I'm fine letting it go though :).

Reply:
Perhaps I didn't understand your code suggestion. Did you mean remove the `self.isWelcomeComplete = onboarding.isWelcomeComplete` from this initializer? If so, then that won't restore the latest state of the onboarding to the transient view model. If not, what specific code change are you proposing?

Reply:
Oh my mistake.  I see now...this is the _other_ direction.  You're initializing this ViewModel with the (initial) value from `onboardingProvider`, but then updating `onboardingProvider` when _this_ ViewModel values' change.  So then the `dropFirst()` is just an optimization...